src/jit/nodeinfo.h

```diff
@@ -122,12 +123,21 @@ class TreeNodeInfo
     unsigned char isTgtPref:1;
     // Whether a spilled second src can be treated as a contained operand
     unsigned char regOptional:1;
+    // Whether or not a node contributes sources to its parent. Used for bookkeeping during
+    // buildRefPositionsForNode in order to avoid unnecessary work.
+    unsigned char contributesSources:1;
```

Comment:
This could use a better explanation. At first I thought this was talking about contained nodes, but then realized that this would be true for any node that produces register result(s), either directly or by passing through the results of its children.